i18n(de): add translation for testing-guide

mikelgo opened the pull request:
#### What kind of changes does this PR include?

German translation for testing guide https://docs.astro.build/de/guides/testing/

hippotastic commented on src/pages/de/guides/testing.md:
```diff
@@ -0,0 +1,120 @@
+i18nReady: false
+setup: import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+---
+Testen hilft dir funktionierenden Astro-Code zu schreiben und zu warten. Astro unterstützt viele beliebte Tools für Unit-Tests, Komponenten-Tests und End-to-End-Tests, darunter Jest, Mocha, Jasmine, Cypress und Playwright. Du kannst sogar Framework-spezifische Testbibliotheken wie React Testing Library installieren, um deine UI-Framework-Komponenten zu testen.
```
```suggestion
Testen hilft dir, funktionierenden Astro-Code zu schreiben und zu warten. Astro unterstützt viele beliebte Tools für Unit-Tests, Komponenten-Tests und End-to-End-Tests, darunter Jest, Mocha, Jasmine, Cypress und Playwright. Du kannst sogar Framework-spezifische Testbibliotheken wie React Testing Library installieren, um deine UI-Framework-Komponenten zu testen.
```

hippotastic commented on src/pages/de/guides/testing.md:
```diff
@@ -0,0 +1,120 @@
+
+### Installation
+
+Du kannst mit der [VS Code Extension](https://playwright.dev/docs/getting-started-vscode) beginnen und deine Tests ausführen.
```
```suggestion
Du kannst mit der [VS Code-Erweiterung](https://playwright.dev/docs/getting-started-vscode) beginnen und deine Tests ausführen.
```

hippotastic commented on src/pages/de/guides/testing.md:
```diff
@@ -0,0 +1,120 @@
+
+Du kannst mit der [VS Code Extension](https://playwright.dev/docs/getting-started-vscode) beginnen und deine Tests ausführen.
+
+Alternativ kannst du Playwright in deinem Astro-Projekt mit dem von dir gewählten Paketmanager installieren. Folge den CLI-Schritten, um Javascript/Typescript, den Namen deines Test Ordners und einen optionalen Github Actions Workflow auszuwählen.
```
```suggestion
Alternativ kannst du Playwright in deinem Astro-Projekt mit dem von dir gewählten Paketmanager installieren. Folge den CLI-Schritten, um JavaScript/TypeScript, den Namen deines Test-Ordners und einen optionalen Github Actions-Workflow auszuwählen.
```

hippotastic commented on src/pages/de/guides/testing.md:
```diff
@@ -0,0 +1,120 @@
+  </Fragment>
+</PackageManagerTabs>
+
+### Erstelle deinen ersten Playwright Test
```
```suggestion
### Erstelle deinen ersten Playwright-Test
```

hippotastic commented on src/pages/de/guides/testing.md:
```diff
@@ -0,0 +1,120 @@
+<html lang="de">
+  <head>
+    <title>Astro ist fantastisch!</title>
+    <meta name='description' content="Beziehe Inhalte von überall und stelle ihn schnell mit Astros next-gen Insel-Architektur bereit." />
```
```suggestion
    <meta name='description' content="Beziehe Inhalte von überall und stelle sie schnell mit Astros Insel-Architektur der nächsten Generation bereit." />
```

hippotastic commented on src/pages/de/guides/testing.md:
````diff
@@ -0,0 +1,120 @@
+
+2. Erstelle einen neuen Ordner und füge nachstehende Testdatei in `src/test` hinzu. Kopiere und füge den nachfolgenden Test in deine Datei ein, um zu verifizieren, dass die Metainformationen der Seite korrekt sind. Aktualisiere den `<title>`-Wert entsprechend der Seite, die du testest.
+
+```jsx title="src/test/index.spec.ts" "Astro is awesome!"
````
```suggestion
```jsx title="src/test/index.spec.ts" "Astro ist fantastisch!"
```

hippotastic commented on src/pages/de/guides/testing.md:
````diff
@@ -0,0 +1,120 @@
+});
+```
+
+:::tip[Eine Basis Url setzen]
````
```suggestion
:::tip[Eine Basis-URL setzen]
```

hippotastic commented on src/pages/de/guides/testing.md:
```diff
@@ -0,0 +1,120 @@
+Du kannst [`"baseURL": "http://localhost:3000"`](https://playwright.dev/docs/api/class-testoptions#test-options-base-url) in der `playwright.config.ts`-Konfigurationsdatei setzen, um `page.goto("/")` anstatt `page.goto("http://localhost:3000/")` als komfortablere URL zu nutzen.
+:::
+
+### Deine Playwright Tests ausführen 
```
```suggestion
### Deine Playwright-Tests ausführen 
```

hippotastic commented on src/pages/de/guides/testing.md:
````diff
@@ -0,0 +1,120 @@
+npx playwright test index.spec.ts
+```
+
+2. Um den vollständigen HTML Testbericht anzusehen, öffne ihn mit folgendem Befehl:
````
```suggestion
2. Um den vollständigen HTML-Testbericht anzusehen, öffne ihn mit folgendem Befehl:
```

hippotastic commented on src/pages/de/guides/testing.md:
```diff
@@ -0,0 +1,120 @@
+
+#### Fortgeschritten: Einen Entwicklungs-Web-Server während deinen Tests starten 
+
+Du kannst Playwright auch deinen Server starten lassen, wenn du dein Test-Skript mit der [`webServer`](https://playwright.dev/docs/test-advanced#launching-a-development-web-server-during-the-tests) Option in der Playwright Konfigurationsdatei ausführst.
```
```suggestion
Du kannst Playwright auch deinen Server starten lassen, wenn du dein Test-Skript mit der [`webServer`](https://playwright.dev/docs/test-advanced#launching-a-development-web-server-during-the-tests)-Option in der Playwright-Konfigurationsdatei ausführst.
```

hippotastic commented on src/pages/de/guides/testing.md:
```diff
@@ -0,0 +1,120 @@
+1. Füge ein Test-Skript wie: `"test:e2e": "yarn playwright"` in deine `package.json`-Datei in der Projektwurzel ein.
+
+2. In `playwright.config.ts` füge das `webServier`-Objekt hinzu und aktualisiere den Kommando-Wert in `yarn preview`.
+3. 
```
```suggestion

```

hippotastic commented on src/pages/de/guides/testing.md:
````diff
@@ -0,0 +1,120 @@
+export default config;
+```
+
+3. Führe `yarn build` und anschließend `yarn test:e2e` aus, um deine Playwright Tests auszuführen.
````
```suggestion
3. Führe `yarn build` und anschließend `yarn test:e2e` aus, um deine Playwright-Tests auszuführen.
```

hippotastic reviewed (commented):
Really high quality translation, thank you! I only found a few grammar mistakes.

Most frequently you are missing the hyphen between compound words. You're in good company though - this is the most common mistake, which is also why it's first in [our translation guide](https://github.com/withastro/docs/tree/main/src/i18n/de#h%C3%A4ufige-fehler). :)

Oh, and it would be great if you could add the translation "Testen" to the navigation bar as well. You need to add an entry for `guides/testing` to [this file](https://github.com/withastro/docs/blob/main/src/i18n/de/nav.ts) to make it work.

Thank you and keep up the good work!

mikelgo commented:
> Really high quality translation, thank you! I only found a few grammar mistakes.
> 
> Most frequently you are missing the hyphen between compound words. You're in good company though - this is the most common mistake, which is also why it's first in [our translation guide](https://github.com/withastro/docs/tree/main/src/i18n/de#h%C3%A4ufige-fehler). :)
> 
> Oh, and it would be great if you could add the translation "Testen" to the navigation bar as well. You need to add an entry for `guides/testing` to [this file](https://github.com/withastro/docs/blob/main/src/i18n/de/nav.ts) to make it work.
> 
> Thank you and keep up the good work!

Thank u very much for your kind words - appreciate it! I fixed all suggestions and added the navigation item as well.

hippotastic reviewed (approved):
LGTM!
